### PR TITLE
Migrate tutorials to main repository, integrate them into documentation

### LINKED ISSUE #2670: Move tutorials to main repository
### Describe the bug

So many questions are written in our issues because people do not seem our tutorials. We need to find a way to export them all to jupyternotebook and put them in the repo. 

### Expected behaviour

People stop asking questions that we have already answered. 

### To Reproduce

_No response_

### Environment Details

_No response_

### Relevant Log Output

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by asumagic**
## What does this PR do?

The goal of this PR is to migrate tutorials from external Colab links to the mainline SpeechBrain repo.

Tutorials will be indexed within the documentation, and will be available in different methods:

- In-documentation preview
- Browser download or GitHub link if too difficult
- Google Colab automatic import

There are multiple upsides to this approach:

- Contributing new tutorials is much less of a pain in the main repo now
- Contributing changes to tutorials is also much nicer (GitHub diffs)
- Tutorials are now significantly easier to find within the documentation or within the repository
- Grouping up tutorials with the documentation makes a lot of sense and is more convenient
- This makes certain keywords indexed in the documentation (and online), which significantly helps searchability
- ~~Linters can help with `.ipynb`, e.g. `codespell` actually works (neat)~~
  - Unfortunately there's no obvious way to stop it from trying to lint cell outputs, which can contain false positives or things that are inconvenient to correct. Multiline regex support exists and would solve this issue although jankily, but only in a currently unreleased version.

Once merged, https://github.com/speechbrain/speechbrain.github.io/pull/22 should be reviewed (this doesn't need to happen immediately after, Colab links won't die).

<details>

<summary>TODO/status</summary>

- [x] Convert the tutorial list to rst documents (done using [this hand rolled script](https://gist.github.com/asumagic/19f9809480b62bfd16094fb5c844a564))
- [x] Integrate them to the toctree
    - ~~That's occupying a pretty big part of the sidebar, maybe it should fold by default?~~ Solved by moving speech task stuff into one category, now it seems reasonable
- [x] Add `.ipynb`s back to main repo
    - [x] **Must** make them smaller to avoid bloating the history prior to any commit
    - [x] Must add a Colab link inside of each of them
    - [x] Use whatever Sphinx integration there is for ipynb
- [x] Replace Colab links with Colab GitHub import links
  - [x] ... and check that those work (they do, although it won't work from the PR as it explicitly references `develop`)
- [x] Add download link (probably through whatever blob URL on GitHub)
- [x] Add in-doc preview
- [x] ... and other files mentioning tutorials there should redirect to the correct documentation page instead.
- [x] Check if it is OK to have a symlink from `tutorials` to `docs/tutorials`
- [x] Ensure that no notebooks/docs make it to the SB wheel
- [x] Ensure that we have no dead links towards tutorials
  - This won't be an issue, I'll just create redirects from existing pages but replace the dropdown in the website to the tutorials/docs page
- [x] Ensure that all toctrees are sane (including within notebooks)
- [x] ~~Ensure that all code blocks are formatted/highlighted correctly (there seem to be issues with this).~~ Seems good enough, TBH
- [x] Remove any cross-references in notebooks to Colab (especially in prerequisites etc.).
- [x] Fix any references to Drive URLs (we migrated to Dropbox for that).
- [x] Fix URLs that aren't detected as such for in-doc tutorials.
- [x] Make summary titles consistent with the actual tutorial name when it didn't match/was modified by me.
- [x] Fix `.rst` heading-level not being applied correctly due to an outdated Sphinx version in RTD. Would updating `docs-requirements.txt` suffice?
- [x] Look into enabling Sphinx RTD extensions explicitly?
- [x] Update descriptions for better formatting.
- [x] ~~Streamline install process instructions and link back to documentation as well.~~ Can do so later, needs testing to do and isn't essential
- [x] Fix LaTeX if possible.
- [x] Probably group up some of the categories, TBH.
- [x] Tutorial dropdown menu on https://speechbrain.github.io should link back to the relevant part of documentation now (see: https://github.com/speechbrain/speechbrain.github.io/pull/22)

</details>

Fixes #2670

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

File: tutorials
Comment (Adel-Moumen) [inline tutorials:1]: in Github, this does not looks that good. Do you think there's a way to directly jump on the folder in `docs/tutorials` when we click on `tutorials/` ?

**Review by Adel-Moumen — APPROVED** [review]
LGTM. This is a very nice and welcome change.

File: tutorials
Comment (asumagic) [reply tutorials:1]: Not that I know of... Tempted to say the symlink should just be removed?

Comment (Adel-Moumen) [reply tutorials:1]: I mean, its better to leave it than nothing right?

**Comment by pplantinga** [comment]
I checked the tutorials in the generated documentation and they look great! Thanks 👍🏻

**Pull request merged**